plugin/advisor/config/config.go

```diff
@@ -0,0 +1,90 @@
+package config
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"gopkg.in/yaml.v3"
+)
+
+//go:embed sql-review.yaml
+var sqlReviewConfigStr string
+
+// MergeSQLReviewRules will merge the input YML config into default template.
+func MergeSQLReviewRules(ymlStr string) ([]*advisor.SQLReviewRule, error) {
+	config := SQLReviewConfiguration{}
+
+	if err := yaml.Unmarshal([]byte(sqlReviewConfigStr), &config); err != nil {
+		return nil, err
+	}
+
+	sqlReviewConfigUpdate := SQLReviewConfigUpdate{}
+	if err := yaml.Unmarshal([]byte(ymlStr), &sqlReviewConfigUpdate); err != nil {
+		return nil, err
+	}
+
+	template := findTemplate(config.TemplateList, sqlReviewConfigUpdate.Template)
+	if template == nil {
+		return nil, fmt.Errorf("cannot find template: %v", sqlReviewConfigUpdate.Template)
+	}
+
+	ruleUpdateMap := make(map[advisor.SQLReviewRuleType]*SQLReviewRuleUpdate)
+	for _, rule := range sqlReviewConfigUpdate.RuleList {
+		ruleUpdateMap[rule.Type] = rule
+	}
+
+	var res []*advisor.SQLReviewRule
+
+	for _, ruleTemplate := range template.RuleList {
+		ruleUpdate := ruleUpdateMap[ruleTemplate.Type]
+		rule, err := mergeRule(ruleTemplate, ruleUpdate)
+		if err != nil {
+			return nil, err
+		}
+		res = append(res, rule)
+	}
+
+	return res, nil
+}
+
+func findTemplate(templateList []*SQLReviewTemplate, id SQLReviewTemplateID) *SQLReviewTemplate {
+	for _, template := range templateList {
+		if template.ID == id {
+			return template
+		}
+	}
+	return nil
+}
+
+func mergeRule(source *SQLReviewTemplateRule, update *SQLReviewRuleUpdate) (*advisor.SQLReviewRule, error) {
```

Comment:
let's write a unit test for this or at a higher level to write a unit test for merging 2 template